File: bridge/java/EmailMessage.java
package Bridge;

// UML diyagramındaki RefinedAbstraction kısmına denk gelmektedir.
// Abstraction (Message) sınıfından türer.
public class EmailMessage extends Message {
    public EmailMessage(IMessageSender messageSender) {
        super(messageSender);
    }

    @Override
    public void send() {
        messageSender.sendMessage(this);
    }
}

File: bridge/kotlin/EmailMessage.kt
package Bridge

// UML diyagramındaki RefinedAbstraction kısmına denk gelmektedir.
// Abstraction (Message) sınıfından türer.
class EmailMessage(messageSender: IMessageSender) : Message(messageSender) {
    override fun send() {
        messageSender.sendMessage(this)
    }
}
